packages/fxa-content-server/app/scripts/lib/experiments/grouping-rules/communication-prefs.js

```diff
@@ -10,22 +10,24 @@
 
 const BaseGroupingRule = require('./base');
 
-const AVAILABLE_LANGUAGES = [
-  'de',
-  'en',
-  'en-[a-z]{2}',
-  'es',
-  'es-[a-z]{2}',
-  'fr',
-  'hu',
-  'id',
-  'pl',
-  'pt-br',
-  'ru',
-  'zh-tw',
-];
+// If only the language, not the region, is specified, then all regions
+// will be considered a match. For instance, 'de' will match 'de', 'de-de',
+// 'de-at', etc. #2217
+const AVAILABLE_LANGUAGES = ['de', 'en', 'es', 'fr', 'hu', 'id', 'pl', 'ru'];
 
-const AVAILABLE_LANGUAGES_REGEX = arrayToRegex(AVAILABLE_LANGUAGES);
+// If the region is specified, other regions will not match. For instance,
+// 'pt-br' will not match 'pt' or 'pt-pt'. #2217
+const AVAILABLE_REGIONS = ['pt-br', 'zh-tw'];
```

Comment:
@shane-tomlinson Note that this change strictly matches languages where the region is specified, like 'pt-br' and 'zh-tw'. These will not match on the base language or other regional languages. The tests have been updated to verify this works as intended.